### PR TITLE
Create lucky money from Mezon #762

### PR DESCRIPTION
**Pull request by phongvu2812**
# Ticket
#762
# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: dong-service/handlers/red_envelope.go [hunk 1]
```diff
@@ -629,6 +629,124 @@ func (r *RedEnvelopeHandler) ClaimRedEnvelopeQR(c *gin.Context) {
...
+		return
+	}
+
+	if !isOwner {
+		logger.Warn().Str("envelope_id", req.ID).Int64("user_id", userID).Msg("User is not the owner")
+		c.JSON(http.StatusBadRequest, models.ErrorResponse(http.StatusBadRequest, constants.ErrUserIDNotMatchRedEnvelopeID))
+		return
+	}
+
+	var statusRedEnvelope string
```
Comment (vietdangquoc) [inline dong-service/handlers/red_envelope.go]: move vào UpdateStatus.
parse status trong UpdateStatus

**Review by vietdangquoc — APPROVED** [review]

File: dong-service/handlers/red_envelope.go [hunk 2]
```diff
@@ -629,6 +629,56 @@ func (r *RedEnvelopeHandler) ClaimRedEnvelopeQR(c *gin.Context) {
...
+
+	logger.Info().
+		Str("envelope_id", redEnvelope.ID).
+		Int64("creator_user_id", userID).
+		Msg("Red envelope created successfully via Mobile")
+
+	c.JSON(http.StatusOK, models.SuccessResponseWithMessage(constants.MsgRedEnvelopeCreated, redEnvelope))
+}
+
+
```
Comment (hoangdohuy) [inline dong-service/handlers/red_envelope.go:786]: Viết 1 hàm kiểm tra nhận vào red-id + txhash
Verify = Zk

Comment (QTR33) [reply dong-service/handlers/red_envelope.go:786]: done
